components/calendar/src/ethiopian.rs

```diff
@@ -131,37 +131,42 @@ impl Calendar for Ethiopian {
     type DateInner = EthiopianDateInner;
     fn date_from_codes(
         &self,
-        era: types::Era,
+        era: Option<types::Era>,
         year: i32,
         month_code: types::MonthCode,
         day: u8,
     ) -> Result<Self::DateInner, DateError> {
-        let year = if era.0 == tinystr!(16, "incar") {
-            if year <= 0 {
-                return Err(DateError::Range {
-                    field: "year",
-                    value: year,
-                    min: 1,
-                    max: i32::MAX,
-                });
-            }
-            year
-        } else if era.0 == tinystr!(16, "pre-incar") {
-            if year <= 0 {
-                return Err(DateError::Range {
-                    field: "year",
-                    value: year,
-                    min: 1,
-                    max: i32::MAX,
-                });
+        let year = if let Some(era) = era {
+            if era.0 == tinystr!(16, "incar") || era.0 == tinystr!(16, "ethiopic") {
+                if year <= 0 {
+                    return Err(DateError::Range {
+                        field: "year",
+                        value: year,
+                        min: 1,
+                        max: i32::MAX,
+                    });
+                }
+                year
+            } else if era.0 == tinystr!(16, "pre-incar")
+                || era.0 == tinystr!(16, "ethiopic-inverse")
+            {
+                if year <= 0 {
+                    return Err(DateError::Range {
+                        field: "year",
+                        value: year,
+                        min: 1,
+                        max: i32::MAX,
+                    });
+                }
+                1 - year
+            } else if era.0 == tinystr!(16, "mundi") || era.0 == tinystr!(16, "ethiopicaa") {
```

Comment:
Nit: `ethioaa` not `ethiopicaa`

Reply:
This isn't what the spec says

Reply:
Updated, but the spec should also be fixed, then.

Reply:
https://github.com/tc39/proposal-intl-era-monthcode/pull/19 , seems like there already was a PR to fix this

Reply:
Yes, I thought that PR was merged long ago but no one ever hit the button.